typescript/sdk/src/consts/environments/index.ts

```diff
@@ -1,32 +1,56 @@
-import { types } from '@hyperlane-xyz/utils';
-
-import { ChainMap } from '../../types';
+import { HyperlaneAgentAddresses } from '../../agents/types';
+import { CoreAddresses } from '../../core/contracts';
+import { IgpAddresses } from '../../gas/contracts';
+import { ChainMap, ChainName } from '../../types';
 import { objMap } from '../../utils/objects';
 
 import mainnet from './mainnet.json';
 import test from './test.json';
 import testnet from './testnet.json';
 
-export const environments = { test, testnet, mainnet };
+export const hyperlaneEnvironments = { test, testnet, mainnet };
+
+export type HyperlaneEnvironment = keyof typeof hyperlaneEnvironments;
+export type HyperlaneEnvironmentChain<E extends HyperlaneEnvironment> = Extract<
+  keyof typeof hyperlaneEnvironments[E],
+  ChainName
+>;
 
-type HyperlaneCoreAddressMap = ChainMap<{
-  mailbox: types.Address;
-  multisigIsm: types.Address;
-  interchainGasPaymaster: types.Address;
-  interchainAccountRouter?: types.Address;
-  interchainQueryRouter?: types.Address;
-  create2Factory: types.Address;
-}>;
+// TODO: Add middleware addresses
+export type HyperlaneContractAddresses = CoreAddresses & IgpAddresses;
 
 // Export developer-relevant addresses
 export const hyperlaneCoreAddresses = objMap(
   { ...testnet, ...mainnet },
   (_chain, addresses) => ({
     mailbox: addresses.mailbox.proxy,
     multisigIsm: addresses.multisigIsm,
+    proxyAdmin: addresses.proxyAdmin,
+    validatorAnnounce: addresses.validatorAnnounce,
+  }),
+) as ChainMap<CoreAddresses>;
+
+export const hyperlaneContractAddresses = objMap(
+  { ...testnet, ...mainnet },
+  (_chain, addresses) => ({
+    mailbox: addresses.mailbox.proxy,
+    multisigIsm: addresses.multisigIsm,
+    proxyAdmin: addresses.proxyAdmin,
+    validatorAnnounce: addresses.validatorAnnounce,
+    interchainGasPaymaster: addresses.interchainGasPaymaster.proxy,
+    storageGasOracle: addresses.storageGasOracle,
+    defaultIsmInterchainGasPaymaster:
+      addresses.defaultIsmInterchainGasPaymaster,
```

Comment:
If it sufficient to call this `defaultInterchainGasPaymaster` or is that misleading?

Reply:
Slightly misleading, unfortunately